src/pages/index.js

```diff
@@ -0,0 +1,310 @@
+
+const previewModal = document.querySelector("#preview-modal");
+const previewModalImage = previewModal.querySelector(".modal__image");
+const previewModalCaption = previewModal.querySelector(".modal__caption");
+const previewModalCloseBtn = previewModal.querySelector(".modal__close-button");
+const cardSubmitButton = cardModal.querySelector(".modal__save-button");
+
+// avatar form element
+const avatarModal = document.querySelector("#avatar-modal");
+const avatarModalBtn = document.querySelector(".profile__avatar-btn");
+const avatarForm = avatarModal.querySelector(".modal__form");
+const avatarModalCloseBtn = avatarModal.querySelector(".modal__close-button");
+const avatarSubmitBtn = avatarModal.querySelector(".modal__close-button");
+const avatarInput = avatarModal.querySelector("#profile-avatar-image");
+
+// delete form element
+const deleteModal = document.querySelector("#delete-modal");
+const deleteForm = deleteModal.querySelector(".modal__form-delete");
+const cancelButton = deleteModal.querySelector(".modal__button-cancel");
+const deleteModalCloseBtn = deleteModal.querySelector(".modal__close-button");
+
+let selectedCard, selectedCardId;
+
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+
+  const cardNameEl = cardElement.querySelector(".card__title");
+  const cardImageEl = cardElement.querySelector(".card__image");
+  const cardLikeButton = cardElement.querySelector(".card__like-button");
+  const isLiked = data.isLiked;
+  if (isLiked) {
+    cardLikeButton.classList.add("card__like-button_liked");
+  }
+  const cardDeleteButton = cardElement.querySelector("#card-delete-button");
+
+  cardNameEl.textContent = data.name;
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
+  cardImageEl.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalImage.src = data.link;
+    previewModalCaption.textContent = data.name;
+    previewModalImage.alt = data.name;
+  });
+
+  cardDeleteButton.addEventListener("click", () =>
+    handleDeleteCard(cardElement, data._id)
+  );
+
+  function handleLike(evt, id) {
+    const isLiked = cardLikeButton.classList.contains(
+      "card__like-button_liked"
+    );
+    api
+      .changeLikeStatus(id, isLiked)
+      .then(() => {
+        cardLikeButton.classList.toggle("card__like-button_liked");
+      })
+      .catch(console.error);
+  }
+  cardLikeButton.addEventListener("click", (evt) => handleLike(evt, data._id));
+
+  return cardElement;
+}
+
+previewModalCloseBtn.addEventListener("click", () => {
+  closeModal(previewModal);
+});
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Deleting...";
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteModal);
+      submitBtn.textContent = "Delete";
```

Comment:
`      submitBtn.textContent = "Delete";
`

You need to **only** return the **default** button text in the `finally` block, so that the button could get the text regardless of whether the request succeeds or fails.